app/models/post.rb

```diff
@@ -37,10 +37,11 @@ def should_generate_new_friendly_id?
   end
 
   def related_posts(limit = 3)
-    Post.joins(:categories)
+    Post
+      .joins(:categories)
       .where(categories: {id: category_ids})
       .where.not(id: id)
       .distinct
-      .limit(limit)
+      .sample(limit)
```

Comment:
does `sample` use SQL to select a random value?

Reply:
No, unfortunately not.
I tried to use `order("RANDOM()")` which was giving me errors and also `offset` in 05ee32229df83f4b458f53bfc16d2b703ebfa322 which was sometimes only giving back two, one or no records at all. I'll have a look at offset again, chances are the generated random number for the offset was just off.

Reply:
what was the error? I guess `order('random()')` should actually work as long as we are using postgres. but for that we would need to make sure the `default_scope` has no effect here (otherwise it would order it by `created_at` and then by random - which would no longer be random as probably no post is created at the same time). 

maybe: 
`Post.unscoped().order('random()').limit(limit)` 
works?

Reply:
Yeah, I think so too. The error was coming from Postgres because the `ORDER_BY` was not place at the correct place in the query.
I'll it with the order you recommended.